tests/python_package_test/test_engine.py

```diff
@@ -50,6 +51,15 @@ def decreasing_metric(preds, train_data):
 def categorize(continuous_x):
     return np.digitize(continuous_x, bins=np.arange(0, 1, 0.01))
 
+def custom_recall(preds, train_data):
+    y_true = train_data.get_label()
+    return 'custom_recall', recall_score(y_true, preds > 0.5), True
+
+
+def custom_precision(preds, train_data):
+    y_true = train_data.get_label()
+    return 'custom_precision', precision_score(y_true, preds > 0.5), True
```

Comment:
Is it possible to use already defined custom metrics in this file? I don't see any tests where metric values are asserted, so I guess `constant_metric`, `decreasing_metric` and others will suit.